src/gpu/intel/jit/gemm/generator/pieces/copy_plan.cpp

```diff
@@ -994,55 +1000,113 @@ void CopyPlan::planEmulatedHFToF4E2M1(CopyInstruction &i)
     ie[8]->src1 = Immediate::uw(8);
 
     // Pack into byte.
-    ie[9]->op = Opcode::and_;
+
+    ie[9]->op = Opcode::shr;
     ie[9]->simd = simd;
     ie[9]->dst = tmp;
     ie[9]->dst.type = DataType::uw;
     ie[9]->src0 = tmp;
     ie[9]->src0.type = DataType::uw;
-    ie[9]->src1 = Immediate(0x00f0);
+    ie[9]->src1 = Immediate::uw(4);
```

Comment:
```suggestion
    ie[8]->src1 = Immediate::uw(12);
```
(and adjust indices below/split size above)